src/request/components/RequestDetailPanelRequest.tsx

```diff
@@ -72,11 +74,34 @@ export class Request extends React.Component<IRequestProps, {}> {
         );
     }
 
-    private renderBody(body: string) {
+    private renderBody(body: string, headers: { [key: string]: string }) {
+        const contentType = this.getContentTypeFromHeaders(headers);
+        const highlightClassName = this.getHighlightClassNameForContentType(contentType);
+
         return (
             <div className='tab-request-body'>
-                <Highlight className=''>{body}</Highlight>
+                <Highlight className={highlightClassName}>{body}</Highlight>
             </div>
         );
     }
+
+    private getContentTypeFromHeaders(headers: { [key: string]: string }): string {
+        let contentType = undefined;
+
+        _.forEach(headers, (value, key) => {
+            if (key.toLowerCase() === 'content-type') {
+                contentType = value;
+                
+                return false;
+            }
+        });
+
+        return contentType;
+    }
```

Comment:
I debated that, since the headers were already provided to the view and highlighting based on content type was also a kind of implementation detail of the view.  However, it's conceivable that we might want to determine the content type in other ways than the `Content-Type` header (e.g. its missing but we know it a priori or we sniff it instead), so it might be most useful to pass it to the view.  I've updated the PR.